hacker/models.py

```diff
@@ -14,92 +20,194 @@
+    - 'last_name' overrides AbstractUser.last_name to require not blank
+    - 'email' overrides AbstractUser.email to require not blank
+
+'''
 
-    # Overrides AbstractUser.last_name to require not blank
+class Hacker(AbstractUser):
+    first_name = models.CharField(max_length=30, blank=False, verbose_name='first name')
     last_name = models.CharField(max_length=150, blank=False, verbose_name='last name')
-    
-    # Overrides AbstractUser.email to require not blank
     email = models.EmailField(blank=False)
 
+    checked_in = models.NullBooleanField(blank=True)
+    email_confirmed = models.BooleanField(blank=True, default=False)
+
+    checked_in_datetime = models.DateTimeField(null=True, blank=True)
+
+    confirm_code = models.CharField(max_length=6, blank=True, null=True)
+
+    def has_related_application(self):
+        a = getattr(self, 'application', None)
+        return a is not None
+
+    def has_related_confirmation(self):
+        c = getattr(self, 'confirmation', None)
+        return c is not None
+
+    def has_related_team(self):
+        c = getattr(self, 'confirmation', None)
+        if c is not None:
+            t = getattr(c, 'team', None)
+            return t is not None
+        else:
+            return False
+
+    def get_related_application(self):
+        a = getattr(self, 'application', None)
+        return a
+
+    def generate_confirm_code(self):
+        code = ''.join(random.choices(string.ascii_uppercase + string.digits, k=settings.EMAIL_CONFIRM_CODE_LENGTH))
+        setattr(self, 'confirm_code', code)
+        self.save()
+
+    def check_confirm_code(self, code):
+        if getattr(self, 'confirm_code', None) is None:
+            return False
+        else:
+            confirm_code = getattr(self, 'confirm_code', None)
+            return (str(confirm_code) == str(code))
+
+    def confirm_email(self, code):
+        if self.check_confirm_code(code):
+            setattr(self, 'email_confirmed', True)
+            setattr(self, 'confirm_code', None)
+            self.save()
+            return True
+        # return 'False' if check_confirm_code(code) returns 'False'
+        else:
+            setattr(self, 'first_name', 'lol nope')
+            return False
+
+    def has_confirmed_email(self):
+        return self.email_confirmed
+
     def __str__(self):
         return '%s, %s' % (self.last_name, self.first_name)
 
 
-class HackerProfile(models.Model):
-    # should only include Hackers w/ verified emails, w/ active status, w/o pre-existing application
-    major = models.CharField(max_length=50)
-    gender = models.CharField(choices=GENDER_CHOICES, max_length=2)
-    classification = models.CharField(choices=CLASSIFICATION_CHOICES, max_length=2)
-    grad_year = models.IntegerField(choices=GRAD_YEAR_CHOICES, verbose_name='graduation year')
-    hacker = models.OneToOneField(          
-        settings.AUTH_USER_MODEL,
-        on_delete=models.CASCADE,
-    )
+'''
```

Comment:
Timestamps/changelogs in comments are generally bad practice, as they become outdated quickly, and even if well-maintained, grow to be too long. Delete